src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -250,12 +252,35 @@ private String syncOnePolicy(AuthenticatedUserRequest userReq,
                                                     IamResourceType iamResourceType,
                                                     UUID resourceId) throws ApiException {
         ResourcesApi samResourceApi = samResourcesApi(userReq.getRequiredToken());
-        List<AccessPolicyResponseEntry> results =
-            samResourceApi.listResourcePolicies(iamResourceType.toString(), resourceId.toString());
-        return results.stream().map(entry -> new PolicyModel()
-            .name(entry.getPolicyName())
-            .members(entry.getPolicy().getMemberEmails()))
-            .collect(Collectors.toList());
+        try (Stream<AccessPolicyResponseEntry> resultStream =
+                 samResourceApi.listResourcePolicies(iamResourceType.toString(), resourceId.toString()).stream()) {
+            return resultStream.map(entry -> new PolicyModel()
+                .name(entry.getPolicyName())
+                .members(entry.getPolicy().getMemberEmails()))
+                .collect(Collectors.toList());
+        }
+    }
+
+    @Override
+    public Map<IamRole, String> retrievePolicyEmails(AuthenticatedUserRequest userReq,
+                                                     IamResourceType iamResourceType,
+                                                     UUID resourceId) throws InterruptedException {
+        SamRetry samRetry = new SamRetry(configurationService);
+        return samRetry.perform(() -> retrievePolicyEmailsInner(userReq, iamResourceType, resourceId));
```

Comment:
It looks like every use of of `SamRetry` is 
```
        SamRetry samRetry = new SamRetry(configurationService);
        return samRetry.perform(() -> [ ... ]);
```
It would be less typing to make this a static method and not bother with a constructor. as it is not used as an object.

Reply:
I'm torn.  It could certainly be one less line.  I'm just trying to think of a case where we may want a handle on the retry object (further logging, etc.)

Reply:
So, I'm not a huge fan of the various "code smell" acronyms and vocabulary.. but here I would say: YAGNI

Reply:
haha.  I had to look that up